core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -651,13 +682,19 @@ public ShareAcquiredRecords acquire(
         lock.writeLock().lock();
         try {
             long baseOffset = firstBatch.baseOffset();
+            // There might be cached batches which are stale due to topic compaction hence archive them.
+            maybeArchiveStaleBatches(fetchOffset, baseOffset);
             // Find the floor batch record for the request batch. The request batch could be
             // for a subset of the in-flight batch i.e. cached batch of offset 10-14 and request batch
             // of 12-13. Hence, floor entry is fetched to find the sub-map.
             Map.Entry<Long, InFlightBatch> floorOffset = cachedState.floorEntry(baseOffset);
```

Comment:
I think I'd call this `floorEntry` because you need to get the key to obtain the offset.

Reply:
Done.